OSDOCS-2457: document procedures for Global Options to Enable HTTP Strict Transport Security (HSTS) (#35812)
openshift/openshift-docs: modules/nw-enabling-hsts.adoc

// * networking/configuring-routing.adoc

[id="nw-enabling-hsts_{context}"]
= HTTP strict transport security

> HTTP Strict Transport Security (HSTS) policy is a security enhancement, which ensures that only HTTPS traffic is allowed on the host. Any HTTP requests are dropped by default. This is useful for ensuring secure interactions with websites, or to offer a secure application for the user's benefit.

Reviewer: ```suggestion
HTTP Strict Transport Security (HSTS) policy is a security enhancement, which signals to the browser client that only HTTPS traffic is allowed on the route's host.  It also optimizes web traffic by signalling HTTPS transport is required, without having to make use of HTTP redirects. HSTS is useful for speeding up interactions with websites.
```

> When HSTS is enforced, HSTS adds a Strict Transport Security header to HTTPS responses from the site. You can use the `insecureEdgeTerminationPolicy` value in a route to redirect to send HTTP to HTTPS. However, when HSTS is enforced, the client changes all requests from the HTTP URL to HTTPS before the request is sent, eliminating the need for a redirect. This is not required to be supported by the client, and can be disabled by setting `max-age=0`.

Reviewer: ```suggestion
When HSTS policy is enforced, HSTS adds a Strict Transport Security header to HTTP/S responses from the site. You can use the `insecureEdgeTerminationPolicy` value in a route to redirect HTTP to HTTPS. However, when HSTS is enforced, the client changes all requests from the HTTP URL to HTTPS before the request is sent, eliminating the need for a redirect.
```

Cluster administrators can:

> * Enforce HSTS per-route

Reviewer: ```suggestion
* Enable HSTS per-route
```

* Disable HSTS per-route

> * Enforce HSTS per-domain

Reviewer: ```suggestion
* Enforce HSTS per-domain, for a set of domains, or use namespace labels in combination with domains
```

[IMPORTANT]
====

> HSTS works only with secure routes, either edge terminated or re-encrypt. The configuration is ineffective on HTTP or passthrough routes.

Reviewer: Some places it is spelled re-encrypt and other place reencrypt.  Whatever is used in other docs is fine.
